src/components/data-loader.jsx

```diff
@@ -36,28 +52,33 @@ const withDataLoader = BaseComponent => (props) => {
     }
   }, [loadMoreItems, onLoadMoreItems]);
 
-  const isBottom = () => {
-    const { scrollHeight, scrollTop, offsetHeight } = ref.current;
-    return scrollHeight - Math.ceil(scrollTop) <= offsetHeight * scrollOffsetFactor;
-  };
-
-  const checkLoadMoreItems = () => {
-    if (!loading && hasMoreData && isBottom(ref)) {
-      setLoadMoreItems(true);
-    }
-  };
-
+  useEffect(() => {
+    const ref = scrollRef.current;
```

Comment:
Wondering whether you need to check if `ref` is not null before continuing?

Reply:
good point. done